tests/htmx/test_middleware.py

```diff
@@ -48,3 +56,42 @@ def view_func():
         result = LoginRequiredMiddleware(lambda x: x).process_view(self.request, view_func, None, {})
         self.assertIsNotNone(result)
         self.assertIsInstance(result, HttpResponseRedirect)
+
+
+class TestHtmxMessageMiddleware(test.TestCase):
+    def setUp(self):
+        request = RequestFactory().get("/foo")
+        request.htmx = True
+        request.user = Mock()
+        self.request = request
+
+        SessionMiddleware(lambda x: x).process_request(self.request)
+        MessageMiddleware(lambda x: x).process_request(self.request)
+        messages.info(self.request, "a message")
+
+        self.middleware = HtmxMessageMiddleware(lambda x: x)
+
+    def process_response(self, response: HttpResponse):
+        return self.middleware.process_response(self.request, response).content.decode()
+
+    def tearDown(self):
+        # expire current messages
+        messages.get_messages(self.request)
+
+    def test_adds_message_to_response(self):
+        self.assertIn("a message", self.process_response(HttpResponse()))
+
+    def test_doesnt_add_message_if_not_htmx(self):
```

Comment:
```suggestion
    def test_doesnt_add_message_to_header_if_not_htmx(self):
```

Because the message is in the request so that plain django will still see messages.

Reply:
changing the test name to `...to_header...`  is wrong. The messages are never added to a header, always the response content. Could do `...to_response...` to be in line with the test_adds_message_to_response test above